### PR TITLE
SBERDOMA-101 refactor firebase auth flow

### PR DESCRIPTION
**Pull request by Dimitreee**

### CODE REVIEW

File: apps/condo/pages/auth/register.tsx [hunk 1]
```diff
@@ -1,75 +1,67 @@
...
     async function sendCode (phoneNumber, recaptchaVerifier) {
-        const phoneProvider = new firebaseAuth.PhoneAuthProvider()
-        const verificationId = await phoneProvider.verifyPhoneNumber(
-            phoneNumber,
-            // @ts-ignore
-            recaptchaVerifier,
-        )
-        return verificationId
+        return await firebaseAuth().signInWithPhoneNumber(phoneNumber, recaptchaVerifier).then((res) => {
+            console.log('res from send conde', res)
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:43]: remove log

File: apps/condo/pages/auth/register.tsx [hunk 2]
```diff
@@ -1,75 +1,67 @@
...
-            .signInWithCredential(credential)
-        // const user = await firebaseAuth()
-        //     .currentUser
-        // // await setCurrentUser(JSON.parse(JSON.stringify(user)))
-        // // const u = await getCurrentUser()
-        // setUser(user)
-        return true
+    async function verifyCode (confirmationResult, verificationCode) {
+        return await confirmationResult.confirm(verificationCode).then(res => {
+            console.log('res from verify code', res)
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:51]: remove log

File: apps/condo/pages/auth/register.tsx [hunk 3]
```diff
@@ -286,44 +275,38 @@ function PhoneAuthForm ({ onPhoneAuthenticated }) {
...
-            setVerificationId(verificationId)
+            setConfirmationResult(confirmationResult)
             verificationCodeTextInput.current?.focus()
         } catch (err) {
             setVerifyError(String(err))
             setVerifyInProgress(false)
+
+            resetRecaptcha()
+            // (Dimitreee): Reset recaptcha verifier if sms didnt send
+            console.log('handle recaptcha re init', err)
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:309]: remove log

**Review by dkoviazin — APPROVED** [review]
Ok

**Review by leonid-d — APPROVED** [review]

**Review by AntonAL — APPROVED** [review]

**Pull request merged**